### DISCUSSION TITLE
do model's attribute get reset to original?

### DISCUSSION THREAD

**Discussion by kobe-ra** [livewire/livewire#3413] [1. Help]
i originally had ma db eloquent call in `mount()` method in livewire component, but then i felt like it is not actually called only once as my attributes were rewritten, so i moved them to laravel's controller.

now in laravel controller
```
public function photos()
{
    $photos = Image::all();

    foreach ($photos as $photo) {
        $photo->imgLqPath = Storage::disk('s3')->temporaryUrl($photo->path025, now()->addHour());
        $photo->imgPath = Storage::disk('s3')->temporaryUrl($photo->path, now()->addHour());
    }

    return view('cs.photos', [
        'photos' => $photos
    ]);
}
```

then pass these photos to the component with
`@livewire('cs.photos', ['photos' => $photos])`

now in component's view i looped them out to see what is happening to them
```
@foreach ($photos as $ph)
<pre>{{ json_encode($ph->getAttributes(), JSON_PRETTY_PRINT|JSON_UNESCAPED_UNICODE) }}</pre>
@endforeach
```

resulting in

```
{
    "id": 204,
    "name": "long_name.png",
    "path": "long_path.png",
    "description": null,
    "path025": "long_path.png",
    "user_id": 5,
    "coord_id": null,
    "gpsx": 12.3,
    "gpsy": 45.56,
    "imageable_type": "App\\CS",
    "imageable_id": 193,
    "created_at": "2021-07-24 14:02:30",
    "updated_at": "2021-07-24 14:02:30",
    "imgLqPath": "https:\/\/path...",
    "imgPath": "https:\/\/path..."
}
```

with this i am happy, however, whenever i call any action via `wire:click` (i also tried `.stop` and `.prevent`.

```
<img
    class="card-img-top"
    src="{{ $photo->imgLqPath }}"
    alt="..."
    height="200px"
    style="object-fit: contain;"
    wire:click="selectPhoto({{ $photo }})"
>
```
(does not matter whether the action does something, problem happens even with blank body of function)
```
public function selectPhoto(Image $photo)
{
    // $this->photo = $photo;
    // $this->emit('photoSelected', 'test');
}
```

my loop for photos attributes changes to this

```
{
    "id": 204,
    "name": "long_name.png",
    "path": "long_path.png",
    "description": null,
    "path025": "long_path.png",
    "user_id": 5,
    "coord_id": null,
    "gpsx": 12.3,
    "gpsy": 45.56,
    "imageable_type": "App\\CS",
    "imageable_id": 193,
    "created_at": "2021-07-24 14:02:30",
    "updated_at": "2021-07-24 14:02:30"
}
```

seems like their attributes were reseted to original, but im clueless why that could happen.

**Comment by kobe-ra** [comment]
okay, so when i call this
```
public function hydrate($value)
{
    dd($value, $this->photos);
}
```

the photo under $value still have path attributes, while `$this->photos` already lost it. which means it happens during hydratation.

the question is why, and how can i stop it?

**Comment by joshhanley** [comment]
@kobe-ra yeah you need to make sure you have a rules array and whitelist your properties that can be updated (such as `imgPath`).

See the docs here for details https://laravel-livewire.com/docs/2.x/properties#binding-models
> Note: For this to work, you have a validation entry in the `$rules` property for any model attributes you want to bind to.

Hope this helps!
